fe/fe-core/src/main/java/com/starrocks/catalog/Type.java

```diff
@@ -149,6 +150,7 @@ public abstract class Type implements Cloneable {
                     .add(DECIMAL64)
                     .add(DECIMAL128)
                     .add(JSON)
+                    .add(UNKNOWN_TYPE)
```

Comment:
why unknown type is added into supported types?

Reply:
> why unknown type is added into supported types?

Because when Fe analzy table, it need to resolve table. All hive types would be converted to sr types, including `UNKNOWN_TYPE`. In code, `ScalarType.createType()` will find `UNKNOWN_TYPE` in `SUPPORTED_TYPES`, if it is removed, exception will be thrown like below.

```
2022-07-26 10:47:24,907 ERROR (starrocks-mysql-nio-pool-1|150) [HiveMetaCache.getTable():315] Failed to get table [databaseName: old_test, tableName: binarytest]
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException: unknown type UNKNOWN_TYPE
    at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2051) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache.get(LocalCache.java:3962) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3985) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4946) ~[spark-dpp-1.0.0.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.getTable(HiveMetaCache.java:313) ~[starrocks-fe.jar:?]
    at com.starrocks.connector.hive.HiveMetadata.getTable(HiveMetadata.java:61) ~[starrocks-fe.jar:?]
    at com.starrocks.server.MetadataMgr.lambda$getTable$1(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
    at java.util.Optional.map(Optional.java:215) ~[?:1.8.0_321]
    at com.starrocks.server.MetadataMgr.getTable(MetadataMgr.java:114) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.resolveTable(QueryAnalyzer.java:669) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.access$100(QueryAnalyzer.java:66) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.resolveTableRef(QueryAnalyzer.java:235) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:161) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitSelect(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.SelectRelation.accept(SelectRelation.java:206) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryRelation(QueryAnalyzer.java:106) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:100) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.visitQueryStatement(QueryAnalyzer.java:83) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:36) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer$Visitor.process(QueryAnalyzer.java:88) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.QueryAnalyzer.analyze(QueryAnalyzer.java:76) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:204) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.visitQueryStatement(Analyzer.java:68) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.QueryStatement.accept(QueryStatement.java:36) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.ast.AstVisitor.visit(AstVisitor.java:104) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer$AnalyzerVisitor.analyze(Analyzer.java:70) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.analyzer.Analyzer.analyze(Analyzer.java:65) ~[starrocks-fe.jar:?]
    at com.starrocks.sql.StatementPlanner.plan(StatementPlanner.java:43) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.StmtExecutor.execute(StmtExecutor.java:346) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:273) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:391) ~[starrocks-fe.jar:?]
    at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:624) ~[starrocks-fe.jar:?]
    at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_321]
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_321]
    at java.lang.Thread.run(Thread.java:750) [?:1.8.0_321]
Caused by: java.lang.NullPointerException: unknown type UNKNOWN_TYPE
    at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:899) ~[spark-dpp-1.0.0.jar:?]
    at com.starrocks.catalog.ScalarType.createType(ScalarType.java:106) ~[starrocks-fe.jar:?]
    at com.starrocks.external.HiveMetaStoreTableUtils.convertColumnType(HiveMetaStoreTableUtils.java:229) ~[starrocks-fe.jar:?]
    at com.starrocks.external.HiveMetaStoreTableUtils.convertToSRTable(HiveMetaStoreTableUtils.java:247) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.loadTable(HiveMetaCache.java:323) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache.access$700(HiveMetaCache.java:39) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:144) ~[starrocks-fe.jar:?]
    at com.starrocks.external.hive.HiveMetaCache$8.load(HiveMetaCache.java:141) ~[starrocks-fe.jar:?]
    at com.google.common.cache.CacheLoader$1.load(CacheLoader.java:192) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[spark-dpp-1.0.0.jar:?]
    at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[spark-dpp-1.0.0.jar:?]
    ... 36 more
```

Reply:
I think you should change the `convertColumnType` method, not here.  I think it's unreasonable we make the UNKNOWN_TYPE as the SUPPORTED_TYPES, but we actually don't support it.

Reply:
> I think you should change the `convertColumnType` method, not here. I think it's unreasonable we make the UNKNOWN_TYPE as the SUPPORTED_TYPES, but we actually don't support it.

In `convertColumnType`, I transfer hive column type to sr column type, so even if unsupported types like hive `Binary` should be given a sr column type.So, `UNKNOWN_TYPE` matches this case.So, `UNKNOWN_TYPE` should be one of `PrimitiveType`.It's truly counter-intuitive, but we still need it.